compilers/concrete-compiler/compiler/include/concretelang/Runtime/workfunction_registry.hpp

```diff
@@ -21,10 +21,8 @@ namespace concretelang {
 namespace dfr {
 
 struct WorkFunctionRegistry;
-namespace {
-static void *dl_handle;
-static WorkFunctionRegistry *_dfr_node_level_work_function_registry;
-} // namespace
+extern WorkFunctionRegistry *_dfr_node_level_work_function_registry;
```

Comment:
Is there a particular reason why this can't simply be a `extern WorkFunctionRegistry` that is defined in a CPP file? It seems odd that the pointer is initialized through the constructor of `WorkFunctionRegistry`.

Reply:
I don't have strong feelings either way. Happy to switch if you prefer.

Reply:
In that case, I think it's OK to keep a pointer (or a smart pointer). However, I think it's best to initialize it from the place that invokes the constructor.